security/compliance_operator/compliance-operator-release-notes.adoc

```diff
@@ -78,6 +93,22 @@ The following advisory is available for the OpenShift Compliance Operator 0.1.59
 
 * In 0.1.57, the Operator started the controller metrics endpoint listening on port `8080`. This resulted in `TargetDown` alerts since cluster monitoring expected port is `8383`. With 0.1.59, the Operator starts the endpoint listening on port `8383` as expected. (link:https://issues.redhat.com/browse/OCPBUGS-3097[*OCPBUGS-3097*])
 
+[id="compliance-operator-0-1-59-known-issue"]
+=== Known issue
+
+In Compliance Operator 0.1.59, the `eventRecordQPS` variable is set to `10` by default, an increase from `5` in previous releases.
```

Comment:
@Vincent056 I didn't quite understand here. What does the "`eventRecordQPS` variable is set to `10` by default mean"? Thanks.
$ cat tp.yaml 
apiVersion: compliance.openshift.io/v1alpha1
kind: TailoredProfile
metadata:
  name: testprofile
  namespace: openshift-compliance
spec:
  description: test
  title: test
  enableRules:
    - name: ocp4-kubelet-configure-event-creation
      rationale: test
    - name: ocp4-version-detect-in-ocp
      rationale: test
$ oc apply -f tp.yaml 
tailoredprofile.compliance.openshift.io/testprofile created
$ oc compliance bind -N test tailoredprofile/testprofile
Creating ScanSettingBinding test
$ oc get suite -w
NAME   PHASE     RESULT
test   RUNNING   NOT-AVAILABLE
test   AGGREGATING   NOT-AVAILABLE
test   DONE          COMPLIANT
test   DONE          COMPLIANT
^C
$ oc get ccr
NAME                                           STATUS   SEVERITY
testprofile-kubelet-configure-event-creation   PASS     medium
$ oc get ccr testprofile-kubelet-configure-event-creation -o=jsonpath={.instructions}
Run the following command on the kubelet node(s):
$ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep eventRecordQPS; done
The output should return .

$ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep eventRecordQPS; done
  "eventRecordQPS": 5,
  "eventRecordQPS": 5,
  "eventRecordQPS": 5,
  "eventRecordQPS": 5,
  "eventRecordQPS": 5,

Reply:
> @Vincent056 I didn't quite understand here. What does the "`eventRecordQPS` variable is set to `10` by default" mean? Thanks.
> 
> ```
> $ cat tp.yaml 
> apiVersion: compliance.openshift.io/v1alpha1
> kind: TailoredProfile
> metadata:
>   name: testprofile
>   namespace: openshift-compliance
> spec:
>   description: test
>   title: test
>   enableRules:
>     - name: ocp4-kubelet-configure-event-creation
>       rationale: test
>     - name: ocp4-version-detect-in-ocp
>       rationale: test
> $ oc apply -f tp.yaml 
> tailoredprofile.compliance.openshift.io/testprofile created
> $ oc compliance bind -N test tailoredprofile/testprofile
> Creating ScanSettingBinding test
> $ oc get suite -w
> NAME   PHASE     RESULT
> test   RUNNING   NOT-AVAILABLE
> test   AGGREGATING   NOT-AVAILABLE
> test   DONE          COMPLIANT
> test   DONE          COMPLIANT
> ^C
> $ oc get ccr
> NAME                                           STATUS   SEVERITY
> testprofile-kubelet-configure-event-creation   PASS     medium
> $ oc get ccr testprofile-kubelet-configure-event-creation -o=jsonpath={.instructions}
> Run the following command on the kubelet node(s):
> $ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep eventRecordQPS; done
> The output should return .
> 
> $ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep eventRecordQPS; done
>   "eventRecordQPS": 5,
>   "eventRecordQPS": 5,
>   "eventRecordQPS": 5,
>   "eventRecordQPS": 5,
>   "eventRecordQPS": 5,
> ```

It means CO checks if the default value for eventRecordQPS is 10 on the cluster.

Reply:
> > @Vincent056 I didn't quite understand here. What does the "`eventRecordQPS` variable is set to `10` by default" mean? Thanks.
> > ```
> > $ cat tp.yaml 
> > apiVersion: compliance.openshift.io/v1alpha1
> > kind: TailoredProfile
> > metadata:
> >   name: testprofile
> >   namespace: openshift-compliance
> > spec:
> >   description: test
> >   title: test
> >   enableRules:
> >     - name: ocp4-kubelet-configure-event-creation
> >       rationale: test
> >     - name: ocp4-version-detect-in-ocp
> >       rationale: test
> > $ oc apply -f tp.yaml 
> > tailoredprofile.compliance.openshift.io/testprofile created
> > $ oc compliance bind -N test tailoredprofile/testprofile
> > Creating ScanSettingBinding test
> > $ oc get suite -w
> > NAME   PHASE     RESULT
> > test   RUNNING   NOT-AVAILABLE
> > test   AGGREGATING   NOT-AVAILABLE
> > test   DONE          COMPLIANT
> > test   DONE          COMPLIANT
> > ^C
> > $ oc get ccr
> > NAME                                           STATUS   SEVERITY
> > testprofile-kubelet-configure-event-creation   PASS     medium
> > $ oc get ccr testprofile-kubelet-configure-event-creation -o=jsonpath={.instructions}
> > Run the following command on the kubelet node(s):
> > $ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep eventRecordQPS; done
> > The output should return .
> > 
> > $ for NODE_NAME in $(oc get nodes -ojsonpath='{.items[*].metadata.name}'); do oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig|.kind="KubeletConfiguration"|.apiVersion="kubelet.config.k8s.io/v1beta1"' | grep eventRecordQPS; done
> >   "eventRecordQPS": 5,
> >   "eventRecordQPS": 5,
> >   "eventRecordQPS": 5,
> >   "eventRecordQPS": 5,
> >   "eventRecordQPS": 5,
> > ```
> 
> It means CO used to have the default variable for checking the eventRecordQPS configuration value was 5 in the past now it is changed to 10.

But the above test was executed with latest compliance operator. If the default variable is 10, should the rule retrun FAIL?

Reply:
I just realized the issue here, we were checking whether eventRecordQPS is equal to or larger than 0 prior to 0.1.57, but then we changed the rule to check if it is matching var_event_record_qps on 0.1.57+.  This may not be a necessary fail check, since different OCP versions might have a different default value.
@sheriff-rh Could you update the notes here? Thanks!

Reply:
If so, this behavior is the same as all the other rules with variables.
Do we really need this PR to clarify this point?

Reply:
I agree with @xiaojiey - if this is not a necessary fail check and different OCP versions can have different values, do we need a note at all? What can we tell Compliance Operator users?